webrender/src/device.rs

```diff
@@ -2015,6 +2180,7 @@ impl<B: hal::Backend> Device<B> {
         self.bound_sampler = [TextureFilter::Linear; 16];
         self.bound_read_fbo = DEFAULT_READ_FBO;
         self.bound_draw_fbo = DEFAULT_DRAW_FBO;
+        self.descriptor_pools.reset();
```

Comment:
we've got to be careful with this. IIRC, reset state is called at least every frame. We are not allowed to re-write the descriptor sets that the GPU is still working on. Resetting the pool here would make us grab a descriptor set that is still in flight.